### DISCUSSION TITLE
Spark NLP 5.0.2: Introducing ONNX Support for ALBERT, CmameBERT, and XLM-RoBERTa, a new Zero-Short Classifier for XLM-RoBERTa transformer, 200+ new ONNX models, and bug fixes!

### DISCUSSION THREAD

**Discussion by maziyarpanahi** [JohnSnowLabs/spark-nlp#13915] [Announcement]
---------------
:loudspeaker: Overview
---------------

Spark NLP 5.0.2 🚀 comes with new ONNX support for ALBERT, CmameBERT, and XLM-RoBERTa annotators, a new Zero-Short Classifier for XLM-RoBERTa transformer, 200+ new ONNX models, and bug fixes! We want to thank our community for their valuable feedback, feature requests, and contributions. Our [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) now contains over 18,000+ free and truly open-source models & pipelines. 🎉

----------------
:fire: New Features
----------------
* **NEW:** Introducing support for ONNX Runtime in `ALBERT`, `CamemBERT`, and `XLM-RoBERTa` annotators. We have already converted 200+ models to ONNX format for these annotators for our community
* **NEW:** Implement `XlmRoBertaForZeroShotClassification` annotator for Zero-Shot multi-class & multi-label text classification based on `XLM-RoBERTa` transformer

----------------
🐛 Bug Fixes & Enhancements
----------------
* Fix MarianTransformers annotator breaking with `java.lang.ClassCastException` in Python
* Fix out of 0.0/1.0 accuracy in SentenceDetectorDL and MultiClassifierDL annotators
* Fix BART issue with a low-temperature value that only occurred when there are no non-infinite logits satisfying the low temperature and top_k values
* Add missing `E5Embeddings` and `InstructorEmbeddings` annotators to `annotators` in Scala for easy all-in-one import

----------------
:book: Documentation
----------------
* [Import models from TF Hub & HuggingFace](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669) 
* [Spark NLP Notebooks](https://github.com/JohnSnowLabs/spark-nlp/tree/master/examples)
* [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) with new models
* [Spark NLP Articles](https://medium.com/spark-nlp)
* [Spark NLP in Action](https://nlp.johnsnowlabs.com/demo)
* [Spark NLP Documentation](https://nlp.johnsnowlabs.com/docs/en/quickstart)
* [Spark NLP Scala APIs](https://nlp.johnsnowlabs.com/api)
* [Spark NLP Python APIs](https://nlp.johnsnowlabs.com/api/python)

----------------
❤️  Community support
----------------

- [Slack](https://join.slack.com/t/spark-nlp/shared_invite/zt-198dipu77-L3UWNe_AJ8xqDk0ivmih5Q) For live discussion with the Spark NLP community and the team
- [GitHub](https://github.com/JohnSnowLabs/spark-nlp) Bug reports, feature requests, and contributions
- [Discussions](https://github.com/JohnSnowLabs/spark-nlp/discussions) Engage with other community members, share ideas,
  and show off how you use Spark NLP!
- [Medium](https://medium.com/spark-nlp) Spark NLP articles
- JohnSnowLabs official [Medium](https://medium.com/john-snow-labs)
- [YouTube](https://www.youtube.com/channel/UCmFOjlpYEhxf_wJUDuz6xxQ/videos) Spark NLP video tutorials

---------------
Installation
---------------
**Python**

```shell
#PyPI

pip install spark-nlp==5.0.2
```

**Spark Packages**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, 3.3.x, and 3.4.x (Scala 2.12):

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp_2.12:5.0.2

pyspark --packages com.johnsnowlabs.nlp:spark-nlp_2.12:5.0.2
```

**GPU**

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:5.0.2

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:5.0.2
```

**Apple Silicon (M1 & M2)**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:5.0.2

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:5.0.2
```

**AArch64**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:5.0.2

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:5.0.2
```

**Maven**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, 3.3.x, and 3.4.x:

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp_2.12</artifactId>
    <version>5.0.2</version>
</dependency>
```

**spark-nlp-gpu:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-gpu_2.12</artifactId>
    <version>5.0.2</version>
</dependency>
```

**spark-nlp-silicon:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-silicon_2.12</artifactId>
    <version>5.0.2</version>
</dependency>
```

**spark-nlp-aarch64:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-aarch64_2.12</artifactId>
    <version>5.0.2</version>
</dependency>
```

**FAT JARs**

* CPU on Apache Spark 3.x/3.1.x/3.2.x/3.3.x/3.4.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-assembly-5.0.2.jar

* GPU on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x/3.4.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-gpu-assembly-5.0.2.jar

* M1 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x/3.4.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-silicon-assembly-5.0.2.jar

* AArch64 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x/3.4.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-aarch64-assembly-5.0.2.jar

## What's Changed
* SPARKNLP-738 Enforcing accuracy to 0 and 1 in classifiers by @danilojsl in https://github.com/JohnSnowLabs/spark-nlp/pull/13901
* Introducing a new Zero-Short Classifier for XLM-RoBERTa transformer by @ahmedlone127 in https://github.com/JohnSnowLabs/spark-nlp/pull/13902
* Add support for ONNX to ALBERT, CamemBERT,  and XLM-RoBERTa by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13907
* SPARKNLP-873 Issue with MarianTransformers models by @danilojsl in https://github.com/JohnSnowLabs/spark-nlp/pull/13908
* BART Bug fix #13898 by @prabod in https://github.com/JohnSnowLabs/spark-nlp/pull/13911
* Models hub by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13913
* release/502-release-candidate by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13912

**Full Changelog**: https://github.com/JohnSnowLabs/spark-nlp/compare/5.0.1...5.0.2

<hr /><em>This discussion was created from the release <a href='https://github.com/JohnSnowLabs/spark-nlp/releases/tag/5.0.2'>Spark NLP 5.0.2: Introducing ONNX Support for ALBERT, CmameBERT, and XLM-RoBERTa, a new Zero-Short Classifier for XLM-RoBERTa transformer, 200+ new ONNX models, and bug fixes!</a>.</em>
